components/RichTextEditor.js

```diff
@@ -302,14 +302,34 @@ export default class RichTextEditor extends React.Component {
       const preText = value.substring(lastIndex, position);
       this.getEditor().setSelectedRange([lastIndex, position]);
       this.getEditor().insertHTML(preText);
-      const attachment = new this.Trix.Attachment({ content: iframe });
+      const { videoService, videoId } = this.getEmbedDetails(iframe);
+      const attachment = new this.Trix.Attachment({
+        content: `<img alt="Preview Image" src="${await this.constructPreviewImageURL(videoService, videoId)}"/>`,
+      });
       this.getEditor().insertAttachment(attachment);
       lastIndex = match.index + iframe.length;
       const postText = value.substring(lastIndex, value.length);
       this.getEditor().insertHTML(postText);
     }
   };
 
+  getEmbedDetails = iframe => {
+    const regex = new RegExp(
+      `(https):\\/\\/([\\w_-]+(?:(?:\\.[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])?`,
+      'ig',
+    );
+    const match = regex.exec(iframe);
+    if (match[0].includes('youtube')) {
+      const matchIdRegex = new RegExp(`embed\\/(.+?)(\\/|$)`, 'ig');
```

Comment:
Could we re-use `parseServiceLink`?

Reply:
@Betree : I don't think so, unless we change that regex (within parseServiceLink). Notice that for example for YouTube embeds; the embed URL such as `https://www.youtube-nocookie.com/embed/JODaYjDyjyQ` isn't recognized as a valid URL by `parseServiceLink`. But the above URL is something that we create (with no cookies); when going to YouTube and if one gets the embed url it will be `https://www.youtube.com/embed/JODaYjDyjyQ` which works with the `parseServiceLink`. Although we can change the `parseServiceLink` url to support no-cookies I didn't do that as it makes the url a bit more complicated and this way seemed simpler. 😉 WDYT?

Reply:
Supporting URLs from youtube-nocookie sounds like something we want to have. Users can eventually be exposed to that URLs if they check the privacy box when sharing: 
![image](https://user-images.githubusercontent.com/1556356/119155172-2b832900-ba53-11eb-8786-f0afb60d05d8.png)

So I would vote in favor of updating the regex in `parseServiceLink` to avoid code duplication, it's a small change:
```diff
- (?:https?://)?(?:www\\.)?youtu(?:\\.be/|be\\.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)
+ (?:https?://)?(?:www\\.)?youtu(?:\\.be/|be(-nocookie)?\\.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)
```

Reply:
@Betree : Ops, that seems simple. I've made this change. 👍🏽